chandra_aca/aca_image.py

```diff
@@ -460,30 +463,43 @@ def get_psf_image(self, row, col, norm=1.0, pix_zero_loc='center'):
         ix = (r + 0.5) / drc - 0.5
         iy = (c + 0.5) / drc - 0.5
 
-        # Int index into PSF library
-        ii = int(floor(ix))
-        jj = int(floor(iy))
-
-        # Following wikipedia notation (Unit Square section of
-        # https://en.wikipedia.org/wiki/Bilinear_interpolation)
-
-        # Float index within subpixel bin in range (0, 1)
-        x = ix - ii
-        y = iy - jj
-
-        # Finally the bilinear interpolation of the PSF images.
-        f = self.psfs
-        b0 = (1 - x) * (1 - y)
-        b1 = x * (1 - y)
-        b2 = (1 - x) * y
-        b3 = x * y
-        P0 = f[ii, jj]
-        P1 = f[ii + 1, jj]
-        P2 = f[ii, jj + 1]
-        P3 = f[ii + 1, jj + 1]
-        psf = P0 * b0 + P1 * b1 + P2 * b2 + P3 * b3
+        if interpolation == 'nearest':
+            # Int index into PSF library
+            ii = int(round(ix))
+            jj = int(round(iy))
+            psf = self.psfs[ii, jj].copy()
+
+        elif interpolation == 'bilinear':
+            # Int index into PSF library
+            ii = int(floor(ix))
+            jj = int(floor(iy))
+
+            # Following wikipedia notation (Unit Square section of
+            # https://en.wikipedia.org/wiki/Bilinear_interpolation)
+
+            # Float index within subpixel bin in range (0, 1)
+            x = ix - ii
+            y = iy - jj
+
+            # Finally the bilinear interpolation of the PSF images.
+            f = self.psfs
+            b0 = (1 - x) * (1 - y)
+            b1 = x * (1 - y)
+            b2 = (1 - x) * y
+            b3 = x * y
+            P0 = f[ii, jj]
+            P1 = f[ii + 1, jj]
+            P2 = f[ii, jj + 1]
+            P3 = f[ii + 1, jj + 1]
+            psf = P0 * b0 + P1 * b1 + P2 * b2 + P3 * b3
+
+        else:
+            raise ValueError("interpolation must be 'nearest' or 'bilinear'")
 
         if norm != 1.0:
             psf *= norm
 
-        return ACAImage(psf, row0=row0, col0=col0)
+        if aca_image:
+            psf = ACAImage(psf, row0=row0, col0=col0)
+
+        return psf
```

Comment:
Would there be any value in also returning row0 and col0 in the non-ACAImage case?